optaplanner-wb-screens/optaplanner-wb-domain-editor/optaplanner-wb-domain-editor-client/src/main/java/org/optaplanner/workbench/screens/domaineditor/client/widgets/planner/PlannerDataObjectEditor.java

```diff
@@ -121,7 +129,20 @@ protected void loadDataObject( DataObject dataObject ) {
         this.comparatorObject = null;
         if ( dataObject != null ) {
             boolean hasPlanningEntity = dataObject.getAnnotation( PlannerDomainAnnotations.PLANNING_ENTITY_ANNOTATION ) != null;
-            boolean hasPlanningSolution = dataObject.getAnnotation( PlannerDomainAnnotations.PLANNING_SOLUTION_ANNOTATION ) != null;
+            boolean hasPlanningSolution = dataObject.getAnnotation( PLANNING_SOLUTION_ANNOTATION ) != null;
+
+            if ( !hasPlanningSolution ) {
+                dataModelerService.call( new RemoteCallback<List<Path>>() {
+                    @Override
+                    public void callback( List<Path> paths ) {
+                        view.enablePlanningSolutionCheckBox( paths.isEmpty() );
```

Comment:
Just an idea.. to disable these before making the async call.. if the call is, for whatever reason "slow", it gives the User the opportunity to click on something they aught not.